RELEASE_PROCESS.md

```diff
@@ -0,0 +1,98 @@
+# Release Process
+
+- [Release Process](#release-process)
+  - [Breaking Changes](#breaking-changes)
+  - [Release Procedure](#release-procedure)
+    - [Creating a new release branch](#creating-a-new-release-branch)
+    - [Cutting a new release](#cutting-a-new-release)
+    - [Tagging Procedure](#tagging-procedure)
+  - [Patch release Procedure](#patch-release-procedure)
+
+This document outlines the release process for Babylon node (babylond)
+
+Babylon follows [semantic versioning](https://semver.org), but with the following deviations to account for state-machine and API breaking changes.
+
+- State-machine breaking changes & API breaking changes will result in an increase of the minor version Y (0.Y.z).
+- All other changes will result in an increase of the patch version Z (0.y.Z).
+
+## Breaking Changes
+
+A change is considered to be ***state-machine breaking*** if it requires a coordinated upgrade for the network to preserve state compatibility
```

Comment:
What about hotfixes that involve state concerns?

Reply:
so here we are bit constrained  by the fact we do not have v1.
I.e gaia in this case (hot fix which may require state fiddling) does minor version bump.

In our case, I would say we need to either:
- release new minor version (if there would not many changes on `main`)
- do a hot-fix which breaks state

I think I would lean to hot-fix which breaks state in such ciriticial situation

Reply:
yup I revised text in `## Patch Release Procedure` to 
```
**Important**: _**Patch releases can break consensus only in exceptional
circumstances .**_
```

as until we have `v1`, there mat be the case that we want to push hot fix which requires coordination. 

After we release `v1`, we can switch this paragraph to it previous version. (i.e patch version must not break consensus)